### DISCUSSION TITLE
Valheim server only starts in debug mode

### DISCUSSION THREAD

**Discussion by TheNieman** [GameServerManagers/LinuxGSM#3239] [General]
The server only starts in debug mode and seems works fine therein. But when I try to run `./vhserver start` I get this:

> [....] Starting vhserver: VHServerno server running on /tmp/tmux-1003/default
> [FAIL] Starting vhserver: Unable to start VHServer

I'm running Ubuntu 20.04.2 on a ACEPC mini PC that's a couple years old.

I googled a bit and thought it might be an issue with tmux, but I'm able to start and exit a tmux session without issue.

**Comment by dgibbs64** [comment]
There have been a number of these issues in the past but I am unsure what causes it. However, the vast majority of servers don't have this issue. So it is something specific to your distro or install of LinuxGSM. Could you provide the output of ./vhserver post-details. It might provide more info that could be useful

**Reply by UndarkAido** [reply]
I've got the same issue. My `mcserver`s were working but stopped after updating a script. They were on `v20.4.1`
```
                          ./+o+-       aidan@Yamato
                  yyyyy- -yyyyyy+      OS: Ubuntu 20.10 groovy
               ://+//////-yyyyyyo      Kernel: x86_64 Linux 5.8.0-43-generic
           .++ .:/++++++/-.+sss/`      Uptime: 21m
         .:++o:  /++++++++/:--:/-      Packages: 2652
        o:+o+:++.`..```.-/oo+++++/     Shell: bash 5.0.17
       .:+o:+o/.          `+sssoo+/    Disk: 1.2T / 9.3T (14%)
  .++/+:+oo+o:`             /sssooo.   CPU: Intel Core i7-2600 @ 8x 3.8GHz [28.0°C]
 /+++//+:`oo+o               /::--:.   RAM: 2015MiB / 7836MiB
 \+/+o+++`o++o               ++////.
  .++.o+++oo+:`             /dddhhh.
       .+.o+oo:.          `oddhhhh+
        \+.++o+o``-````.:ohdhhhhh+
         `:o+++ `ohhhhhhhhyo++os:
           .o:`.syhhhhhhh/.oo++o`
               /osyyyyyyo++ooo+++/
                   ````` +oo+++o\:
                          `oo++.
```
```
$ ./vhserver dt

Distro Details
=================================
Distro:    Ubuntu 20.10
Arch:      x86_64
Kernel:    5.8.0-43-generic
Hostname:  Yamato
Uptime:    0d, 0h, 23m
tmux:      tmux 3.1b
glibc:     2.32

Server Resource
=================================
CPU
Model:      Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz
Cores:      8
Frequency:  1627.836MHz
Avg Load:   0.27, 0.34, 0.27

Memory
Mem:       total  used   free   cached  available
Physical:  7.7GB  1.8GB  5.7GB  2.0GB   5.7GB
Swap:      2.0GB  0B     2.0GB

Storage
Filesystem:  /dev/sda1
Total:                  220G
Used:                   84G
Available:   125G

Network
IP:           0.0.0.0
Internet IP:  66.68.80.89

Game Server Resource Usage
=================================
CPU Used:  0%
Mem Used:  0%  0MB

Storage
Total:        29G
Serverfiles:  997M
Backups:      128K

Valheim Server Details
=================================
Server name:      Aidan'sValheim
App ID:           896660
Server IP:        0.0.0.0:2456
Internet IP:      66.68.80.89:2456
Server password:  XXX
Game world:       vhserver
Status:           OFFLINE

vhserver Script Details
=================================
Script name:            vhserver
LinuxGSM version:       v21.1.3
glibc required:         2.15
Discord alert:          off
Email alert:            off
IFTTT alert:            off
Mailgun (email) alert:  off
Pushbullet alert:       off
Pushover alert:         off
Rocketchat alert:       off
Slack alert:            off
Telegram alert:         off
Update on start:        off
User:                   lgsm
Location:               /home/lgsm

Backups
=================================
No Backups created

Command-line Parameters
=================================
 ./valheim_server.x86_64 -name Aidan'sValheim -password XXX -port 2456 -world vhserver -public 0

Ports
=================================
Change ports by editing the parameters in:
/home/lgsm/lgsm/config-lgsm/vhserver

Useful port diagnostic command:
netstat -atunp | grep valheim

DESCRIPTION  DIRECTION  PORT  PROTOCOL
> Game       INBOUND    2456  udp
> Query      INBOUND    2457  udp

Status: OFFLINE
```

**Reply by UndarkAido** [reply]
Is there anything else you need from me @dgibbs64? I'd really like to be able to run my servers properly again.

**Reply by wldwalker** [reply]
It's because you have an apostrophe in the game server name I think. Valheim doesn't like it when you use special characters in the server name. I believe an update is in the pipeline to address this.

**Reply by UndarkAido** [reply]
> It's because you have an apostrophe in the game server name I think. Valheim doesn't like it when you use special characters in the server name. I believe an update is in the pipeline to address this.

Oups well I've got to change that password now lol.

That doesn't explain my `mcserver` that's now failing with the same error or how both servers run fine in `debug` mode. Sorry I left both those bits out,I think I forgot because I'd brought it up in Discord. 

```

Distro Details
=================================
Distro:    Ubuntu 20.10
Arch:      x86_64
Kernel:    5.8.0-43-generic
Hostname:  Yamato
Uptime:    6d, 16h, 5m
tmux:      tmux 3.1b
glibc:     2.32

Server Resource
=================================
CPU
Model:      Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz
Cores:      8
Frequency:  2313.837MHz
Avg Load:   0.73, 0.72, 0.67

Memory
Mem:       total  used   free   cached  available
Physical:  7.7GB  3.4GB  4.1GB  2.9GB   4.1GB
Swap:      2.0GB  698MB  1.4GB

Storage
Filesystem:  /dev/sda1
Total:                  220G
Used:                   120G
Available:   88G

Network
IP:           0.0.0.0
Internet IP:  72.179.21.5

Game Server Resource Usage
=================================
CPU Used:  0%
Mem Used:  0%  0MB

Storage
Total:        29G
Serverfiles:  3.2G
Backups:      128K

Minecraft Server Details
=================================
Server name:    LinuxGSM
Branch:         release
Server IP:      0.0.0.0:25565
Internet IP:    X.X.X.X:25565
Query enabled:  true
RCON password:  adminVzGdBVP0
Maxplayers:     20
Game mode:      NOT SET
Game world:     world
Status:         OFFLINE

mcserver-direwolf Script Details
=================================
Script name:            mcserver-direwolf
LinuxGSM version:       v21.1.3
Discord alert:          off
Email alert:            off
IFTTT alert:            off
Mailgun (email) alert:  off
Pushbullet alert:       off
Pushover alert:         off
Rocketchat alert:       off
Slack alert:            off
Telegram alert:         off
Update on start:        off
User:                   lgsm
Location:               /home/lgsm
Config file:            /home/lgsm/serverfiles-direwolf/server.properties

Backups
=================================
No Backups created

Command-line Parameters
=================================
java -Xmx1024M -jar java -server -XX:+UseG1GC -Xmx3G -Xms3G -Dsun.rmi.dgc.server.gcInterval=2147483646 -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M -jar /home/lgsm/serverfiles-direwolf/minecraft_server.jar nogui

Ports
=================================
Change ports by editing the parameters in:
/home/lgsm/serverfiles-direwolf/server.properties

Useful port diagnostic command:
netstat -atunp | grep java

DESCRIPTION  DIRECTION  PORT   PROTOCOL
> Game       INBOUND    25565  tcp
> Query      INBOUND    25565  udp
> Rcon       INBOUND    25575  tcp

Status: OFFLINE
```

**Reply by Janestar80** [reply]
Thanks a ton @buer22 - that actually sorted it for me!

**Comment by nickmayer** [comment]
Same thing is happening to me. No special characters anywhere. Discord suggested it may be tmux related so I tried 3 different version (2.8, 3.0b, 3.2) without any changes to outcome.

**Comment by nickmayer** [comment]
Looks like it is an issue with the call to tmux in command_start.sh. Something is not-quite-right with the quoting. If I remove all double quotes from the line it works fine.

It was attempting to run `tmux new-session -d -x "80" -y "23" -s "vhserver" " ./valheim_server.x86_64 -name 'LastAcorn' -password nutzz -port 2456 -world vhserver -public 0"`. I suspect tmux was trying to treat the command as single filename with spaces include rather than a command with arguments. For example `tmux new-session -s test "sleep 5"` fails locally because there is no executable named `"sleep 5"`. Whereas `tmux new-session -s test sleep 5` works as expected.

**Comment by marcleblanc2** [comment]
I was stumped on this one for a few hours. This is what worked for me, in the end, I removed the double quotes from the command on [command_start.sh#L75](https://github.com/GameServerManagers/LinuxGSM/blob/532443eafeee33706399a980070be70e2d5693fd/lgsm/functions/command_start.sh#L75)

tmux new-session -d -x ${sessionwidth} -y ${sessionheight} -s ${sessionname} ${preexecutable} ${executable} ${startparameters} 2> "${lgsmlogdir}/.${selfname}-tmux-error.tmp"

Borrowed the idea from @nickmayer above. 

Running Valheim, on
Ubuntu, 20.04 LTS, amd64 focal image build on 2021-03-15
ldd (Ubuntu GLIBC 2.31-0ubuntu9.2) 2.31
tmux 3.0a
LinuxGSM - Valheim - Version v21.2.2

**Reply by Pitabred** [reply]
This got me going, too! FYI, the only thing you really need to remove the quotes from is the section after "-s", the others can (and probably should) stay quoted

**Comment by marcleblanc2** [comment]
I opened a new issue for this: #3397
